projects/components/src/table/cells/data-parsers/table-cell-string-parser.ts

```diff
@@ -1,16 +1,35 @@
+import { MetricAggregation } from '@hypertrace/distributed-tracing';
 import { TableCellParser } from '../table-cell-parser';
 import { TableCellParserBase } from '../table-cell-parser-base';
 import { CoreTableCellParserType } from '../types/core-table-cell-parser-type';
 
 @TableCellParser({
   type: CoreTableCellParserType.String
 })
-export class TableCellStringParser extends TableCellParserBase<string, string, string> {
-  public parseValue(cellData: string): string {
-    return cellData;
+export class TableCellStringParser extends TableCellParserBase<string, CellValue, CellValue> {
+  public parseValue(cellData: CellData): CellValue {
+    return this.extractValue(cellData);
   }
 
-  public parseFilterValue(cellData: string): string {
+  private extractValue(cellData: CellData): CellValue {
+    switch (typeof cellData) {
+      case 'string':
+        return cellData;
+      case 'object':
+        return cellData?.value ?? undefined;
+      default:
+        return undefined;
```

Comment:
I think it would be safer to make this return cellData in cases other than `cellData?.value` - that way, it should be more or less backwards compatible. Here, if it previously returned a number for example, it would have been correctly rendered as text. Now it would be turned to undefined. Alternatively, we could use `String()` to stringify values (although I'd still pass undefined and null through for the renderer to deal with)

Reply:
makes sense